Source/Core/TopLevelEncoder.swift

```diff
@@ -0,0 +1,7 @@
+import Foundation
+
+protocol TopLevelEncoder {
```

Comment:
same as https://github.com/checkout/frames-ios/pull/85/files#r672252221

What is a `TopLevelEncoder` and does it make more sense than `Encoding`

Reply:
Responded above.